lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -623,40 +623,81 @@ int amf_app_handle_pdu_session_response(
 
   smf_ctx->n_active_pdus += 1;
 
-  if (REGISTERED_IDLE == ue_context->mm_state) {
-    // pdu session state
-    smf_ctx->pdu_session_state            = ACTIVE;
-    amf_sap_t amf_sap                     = {};
-    amf_sap.primitive                     = AMFAS_ESTABLISH_CNF;
-    amf_sap.u.amf_as.u.establish.ue_id    = ue_id;
-    amf_sap.u.amf_as.u.establish.nas_info = AMF_AS_NAS_INFO_SR;
-
-    amf_sap.u.amf_as.u.establish.pdu_session_status_ie =
-        (AMF_AS_PDU_SESSION_STATUS | AMF_AS_PDU_SESSION_REACTIVATION_STATUS);
-    amf_sap.u.amf_as.u.establish.pdu_session_status =
-        (1 << smf_ctx->smf_proc_data.pdu_session_id);
-    amf_sap.u.amf_as.u.establish.pdu_session_reactivation_status =
-        (1 << smf_ctx->smf_proc_data.pdu_session_id);
-    amf_sap.u.amf_as.u.establish.guti = ue_context->amf_context.m5_guti;
-    rc                                = amf_sap_send(&amf_sap);
-    if (RETURNok == rc) {
-      ue_context->mm_state = REGISTERED_CONNECTED;
+  if (!ue_context->pending_service_response) {
+    if (REGISTERED_IDLE == ue_context->mm_state) {
+      // pdu session state
+      smf_ctx->pdu_session_state            = ACTIVE;
+      amf_sap_t amf_sap                     = {};
+      amf_sap.primitive                     = AMFAS_ESTABLISH_CNF;
+      amf_sap.u.amf_as.u.establish.ue_id    = ue_id;
+      amf_sap.u.amf_as.u.establish.nas_info = AMF_AS_NAS_INFO_SR;
+
+      amf_sap.u.amf_as.u.establish.pdu_session_status_ie =
+          (AMF_AS_PDU_SESSION_STATUS | AMF_AS_PDU_SESSION_REACTIVATION_STATUS);
+      amf_sap.u.amf_as.u.establish.pdu_session_status =
+          (1 << smf_ctx->smf_proc_data.pdu_session_id);
+      amf_sap.u.amf_as.u.establish.pdu_session_reactivation_status =
+          (1 << smf_ctx->smf_proc_data.pdu_session_id);
+      amf_sap.u.amf_as.u.establish.guti = ue_context->amf_context.m5_guti;
+      rc                                = amf_sap_send(&amf_sap);
+      if (RETURNok == rc) {
+        ue_context->mm_state = REGISTERED_CONNECTED;
+      }
+    } else {
+      OAILOG_DEBUG(
+          LOG_AMF_APP,
+          "Sending message to gNB for PDUSessionResourceSetupRequest "
+          "**n_active_pdus=%d **\n",
+          smf_ctx->n_active_pdus);
+
+      amf_smf_msg.pdu_session_id = pdu_session_resp->pdu_session_id;
+      /*Execute PDU establishement accept from AMF to gnodeb */
+      pdu_state_handle_message(
+          // ue_context->mm_state, STATE_PDU_SESSION_ESTABLISHMENT_ACCEPT,
```

Comment:
Please remove commented lines if not required.

Reply:
Done